ipc/api/src/checkpoint.rs

```diff
@@ -86,3 +96,81 @@ pub struct BottomUpCheckpoint {
     /// The list of messages for execution
     pub msgs: Vec<IpcEnvelope>,
 }
+
+fn deserialize_vec_bytes_from_vec_hex<'de, D>(
+    deserializer: D,
+) -> anyhow::Result<Vec<Vec<u8>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let data = <Vec<String>>::deserialize(deserializer)?;
+
+    let mut ret = Vec::new();
+    for s in data {
+        let s =
+            hex::decode(&s).map_err(|_| D::Error::custom(format!("cannot decode hex: {}", s)))?;
+        ret.push(s);
+    }
+    Ok(ret)
+}
+
+pub fn serialize_vec_bytes_to_hex<T: AsRef<[u8]>, S>(data: &[T], s: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let values = data.iter().map(hex::encode).collect::<Vec<_>>();
```

Comment:
It doesn't looks like there is a need to collect into a `Vec`, you can just iterate over the elements and hex them as you go, no?